addons/mail_client_extension/controllers/main.py

```diff
@@ -0,0 +1,323 @@
+    def res_partner_get_by_email(self, email, name, **kwargs):
+        response = {
+            'authorized': True,
+        }
+
+        #compute the sender's domain
+        normalized_email = tools.email_normalize(email)
+        if not normalized_email:
+            response['error'] = 'Bad email.'
+            return response
+        sender_domain = normalized_email.split('@')[1]
+
+        # Search for the partner based on the email.
+        # If multiple are found, take the first one.
+        partner = request.env['res.partner'].with_user(request.env.user).search([('email', '=', email)], limit=1)
+        if partner:
+            response['partner'] = {
+                'id': partner.id,
+                'name': partner.name,
+                'title': partner.function,
+                'email': partner.email,
+                'image': partner.image_128,
+                'phone': partner.phone,
+                'mobile': partner.mobile,
+                'company': {},
+                'enrichment_info': None
+            }
+            # if there is already a company for this partner, just take it without enrichment.
+            if partner.parent_id:
+                response['partner']['company'] = self._get_company_dict(partner.parent_id)
+            else:
+                company = self._find_existing_company(sender_domain)
+                if not company: # create and enrich company
+                    company, enrichment_info = self._create_company_from_iap(sender_domain)
+                    response['enrichment_info'] = enrichment_info
+                response['partner']['company'] = self._get_company_dict(company)
+        else: #no partner found
+            response['partner'] = {
+                'id': -1,
+                'name': name,
+                'email': email,
+                'company': {},
+                'enrichment_info': None
+            }
+            company = self._find_existing_company(sender_domain)
+            if not company:  # create and enrich company
+                company, enrichment_info = self._create_company_from_iap(sender_domain)
+                response['enrichment_info'] = enrichment_info
+            response['partner']['company'] = self._get_company_dict(company)
+
+        return response
+
+    @http.route('/mail_client_extension/partner/create', type="json", auth="outlook", csrf=False, cors="*")
+    def res_partner_create(self, email, name, company, **kwargs):
+        # TODO search the company again instead of relying on the one provided here?
+
+        # Create the partner if needed.
+        partner_info = {
+            'name': name,
+            'email': email,
+        }
+        if company > -1:
+            partner_info['parent_id'] = company
+        partner = request.env['res.partner'].with_user(request.env.user).create(partner_info)
+
+        response = {
+            'authorized': True,
+            'created': True,
```

Comment:
`authorized` seems outdated, `created` seems unnecessary as there's no situation in which it'll return a success without having created the partner?